patroni/postgresql.py

```diff
@@ -223,24 +223,72 @@ def sync_from_leader(self, leader):
         r = parseurl(leader.conn_url)
 
         env = self.write_pgpass(r)
-        return self.create_replica(r, env) == 0
+        ret = self.create_replica(leader, env) == 0
+        ret and self.delete_trigger_file()
+        return ret
 
     @staticmethod
     def build_connstring(conn):
-        return "host={host} port={port} user={user}".format(**conn)
+        """
+        >>> Postgresql.build_connstring({'host': '127.0.0.1', 'port': '5432'}) == 'host=127.0.0.1 port=5432 '
+        True
+        """
+        mconn = ""
+        for param, val in sorted(conn.items()):
+            mconn = mconn + "{0}={1} ".format(param, val)
```

Comment:
Do we really need this awkward space at the end?
And actually we can have one-liner here:

```python
return ' '.join('{}={}'.format(param, val) for param, val in sorted(conn.items()))
```